tests/Query/BuilderTest.php

```diff
@@ -76,6 +76,19 @@ public static function provideQueryBuilderToMql(): iterable
             fn (Builder $builder) => $builder->limit(10)->offset(5)->select('foo', 'bar'),
         ];
 
+        yield 'where accepts $ in operators' => [
+            ['find' => [
+                ['$or' => [
+                    ['foo' => ['$type' => 2]],
+                    ['foo' => ['$type' => 4]],
```

Comment:
Before the patch, the result was `['foo' => '$type']` because `Builder::invalidOperator`  is true and [transforms](https://github.com/laravel/framework/blob/10.x/src/Illuminate/Database/Query/Builder.php#L799-L801) the filter to use `$operator` as `$value`.